scripts/prepare.ts

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable no-console */
+import path from 'path';
+import * as shell from 'shelljs';
+import chalk from 'chalk';
+import fse from 'fs-extra';
+import { PackageJson } from 'type-fest';
+
+const { hashElement } = require('folder-hash');
+
+const { babelify } = require('./utils/compile-babel');
+const { tscfy, downgrade } = require('./utils/compile-tsc');
+
+interface Info {
+  modulePath: string;
+  packagePath: string;
+  checksumPath: string;
+  package: PackageJson;
+  checksum: {
+    checksum: string;
+  };
+}
+
+async function getInfo(): Promise<Info> {
+  const modulePath = path.resolve('./');
+  const packagePath = path.join(modulePath, 'package.json');
+  const checksumPath = path.join(modulePath, 'dist', 'checksum.json');
+
+  const exists = await fse.pathExists(checksumPath);
+
+  if (!exists) {
+    await fse.ensureFile(checksumPath);
+    await fse.writeJSON(checksumPath, {});
+  }
+
+  return {
+    modulePath,
+    packagePath,
+    checksumPath,
+    package: await fse.readJSON(packagePath),
+    checksum: await fse.readJSON(checksumPath),
+  };
+}
+
+async function createChecksum({ modulePath }: Info) {
+  const { hash } = await hashElement(path.join(modulePath, 'src'), {
+    folders: { exclude: ['tests', '__mocks__', '__tests__'] },
+    files: {
+      include: ['*.js', '*.jsx', '*.ts', '*.jsx', '*.json'],
```

Comment:
**jsx is here twice. Missing tsx.**

Random question, is worth to create 2 hashes, one for src, one for ts, tsx which would only do tsc when this second hash changes.

It may add another level of complexity to this and maybe not needed !